pkg/aws/aws.go

```diff
@@ -0,0 +1,144 @@
+package aws
+
+import (
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/autoscaling/autoscalingiface"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	// TODO: support multiple ASG
+	ASClient         autoscalingiface.AutoScalingAPI
+	AsgName          string
+	LaunchConfigName string
+}
+
+func (c *Client) AnASGNamed(name string) error {
+
+	sess, err := GetAWSCredentials()
+	if err != nil {
+		return errors.Errorf("Failed getting AWS credentials: %v", err)
+	}
+
+	ASC := autoscaling.New(sess)
```

Comment:
Should we move this out? Maybe we should put this inside `GetAWSCredentials`, otherwise you would have to do this for every step implementation around ASG

Reply:
I am not quite sure I understand your point. What do you mean by that?
>otherwise you would have to do this for every step implementation around ASG

Reply:
If the client is part of the Client part, why can't we use c.ASClient instead of instantiating and setting the client at the bottom:
```
c.ASClient = ASC
```

Should we do this work (instantiate and cache the client) inside `GetAWSCredentials` instead?

Reply:
Right now `AnASGNamed` is used to set the current ASG and fetch the AS client. From there, the other methods can be use and they will act on the current ASG.

If we set the AS client in the `GetAWSCredentials ` it should return nothing and we could rename it `GetAWSClient`. But then `AnASGNamed` sort of loses its purpose I believe.

Let me know what you think

Reply:
Implemented what we discussed in slack.